lib/ansible/modules/windows/win_product_facts.ps1

```diff
@@ -0,0 +1,48 @@
+#!powershell
+# This file is part of Ansible
+
+# Copyright: (c) 2017, Dag Wieers (dagwieers) <dag@wieers.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+Function Get-ProductKey {
+    # First try to find the product key from ACPI
+    $product_key = (Get-CimInstance -Class SoftwareLicensingService).OA3xOriginalProductKey
```

Comment:
Probably also want a `-ErrorAction Continue` here in case the CIM class or property is not defined (eg, Server 2008)

Reply:
But then you cannot get to the OA3xOriginalProductKey property either. So should it be a try/catch instead ?